### DISCUSSION TITLE
设计师反馈加上 这个pag 后画面变得黑漆漆

### DISCUSSION THREAD

**Discussion by wang-sy** [Tencent/libpag#1736] [Q&A] [answered]
[](url)设计师反馈现在渲染出来的画面黑漆漆的，和预览效果不同

> 需要一层alpha预乘的操作来保持半透明区域的颜色抠黑

但是我理解我们这边是没法通过代码来调整具体的 alpha 通道的属性？

所以可能需要帮忙看下这个 pag 文件是否存在问题？

[氛围.pag.zip](https://github.com/Tencent/libpag/files/12655670/pag.zip)

**Comment by domchen** [comment, marked as answer]
你这个文件就是一个视频序列帧，带了透明通道。没有任何复杂结构。你是打算渲染出来逐帧盖在画面上对吧？这个取决于你怎么盖上去的方式。这个pag文件渲染出来的内容就是带了半透明的，直接正常normal方式混合上去就行，不需要什么额外操作。如果不符合预期，问题不出在文件上，出在具体混合的地方。因为你文件里也没有占位图，没法替换，只可能是外部做的混合，那就要从外部检查，PAG没做其他的操作。
